include/fastdds/dds/topic/TopicDataType.hpp

```diff
@@ -271,9 +239,9 @@ class TopicDataType
 private:
 
     //! Data Type Name.
-    std::string m_topicDataTypeName;
+    std::string topic_data_typename_;
     //TODO(XTypes)
-    bool auto_fill_type_information_;
+    bool auto_fill_type_information_ {true};
 
     friend class fastdds::dds::TypeSupport;
```

Comment:
Since the private members have getter and setter methods, I think we can remove this friend